pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -78,9 +78,43 @@ func (ir *IngressReconciler) Reconcile(ctx context.Context, isvc *v1beta1.Infere
 	serviceHost := getServiceHost(isvc)
 	serviceUrl := getServiceUrl(isvc, ir.ingressConfig)
 	disableIstioVirtualHost := ir.ingressConfig.DisableIstioVirtualHost
+
+	// If the "serving.kserve.io/stop" annotation is present in isvc, follow the force stop logic
+	forceStopRuntime := "false"
+	if val, exist := isvc.Annotations[constants.StopAnnotationKey]; exist {
+		forceStopRuntime = val
+	}
```

Comment:
There seems to be more than one instance of this code, and not all instances are equal. The other one is using `forceStopRuntime = strings.EqualFold(val, "true")` (although I see the following conditional is using the EqualFold helper)

I suggest to extract these lines to a func/method in the `InferenceService` struct.